### DISCUSSION TITLE
Why we should verify HTTP response bodies, and why we should use zod for this

### DISCUSSION THREAD

**Discussion by giscus** [timdeschryver/timdeschryver.dev#332] [Announcements]
# Why we should verify HTTP response bodies, and why we should use zod for this

Let's explore zod together and see it can be useful to create a safer enviroment by parsing HTTP response bodies.

https://timdeschryver.dev/blog/why-we-should-verify-http-response-bodies-and-why-we-should-use-zod-for-this

**Comment by EremenkoAndrey** [comment]
Thank you for the useful article.
I have found solution with more effective type inference. I hope it will helpful to somebody.

```
function parseResponse<T>(schema: ZodType<T>): OperatorFunction<Object, T> {
  return input$ => input$.pipe(
    tap((value: any) => {
      return schema.parse(value);
    })
  );
}
```

**Reply by timdeschryver** [reply]
Thanks @EremenkoAndrey !

**Comment by Vanotera** [comment]
Nicely explained; It's very clear to me now how to implement zod in my current project!
Thanks!
